### ISSUE TITLE
GLB_CENTROIDS grid datasets and NAT_REG_ID

### ISSUE THREAD

**Issue by tovogt** [CLIMADA-project/climada_python#31]
There are two global grid datasets in the CLIMADA repository, but there is no documentation about how they have been generated. One grid is in 150 arc-seconds resolution (`GLB_CENTROIDS_NC`). The other grid is in 360 arc-seconds resolution and does only contain grid points on land (`GLB_CENTROIDS_MAT`). Both contain grid points together with country codes:
https://github.com/CLIMADA-project/climada_python/blob/1d8aad6c7c4a021af09de561733f5aafae5c02cf/climada/util/constants.py#L54-L58
They come together with a look-up table for the ISO 3166 alpha-3 codes of the country codes used:
https://github.com/CLIMADA-project/climada_python/blob/1d8aad6c7c4a021af09de561733f5aafae5c02cf/climada/util/constants.py#L73-L74
The country codes are **not** in accordance with country codes in Natural Earth. For example, they contain the region "Serbia and Montenegro" which is actually split up since 2006, same for "Netherlands Antilles" (split up since 2010).

### 1) Replace old grids
First, I propose to remove the two datasets `GLB_CENTROIDS_NC` and `GLB_CENTROIDS_MAT` in favor of the more recent `NATEARTH_CENTROIDS_150AS` and `NATEARTH_CENTROIDS_360AS`:
https://github.com/CLIMADA-project/climada_python/blob/1d8aad6c7c4a021af09de561733f5aafae5c02cf/climada/util/constants.py#L60-L66
The new grids come together with a generation function for reproducibility:
https://github.com/CLIMADA-project/climada_python/blob/1d8aad6c7c4a021af09de561733f5aafae5c02cf/climada/hazard/centroids/centr.py#L825-L832
I managed to replace all occurrences of the old grids in the current develop branch, and they are actually not needed anywhere. Furthermore, the data files of the newer grids are smaller and in the HDF5 format used by the `Centroids` class.

The file `GLB_CENTROIDS_NC` is part of the ISIMIP project: https://www.isimip.org/gettingstarted/input-data-bias-correction/details/13/ It is based on the GPWv3 "National Identifier Grid": https://sedac.ciesin.columbia.edu/data/set/gpw-v3-national-identifier-grid Maybe we could download it from ISIMIP or from SEDAC instead of including it in CLIMADA?

### 2) Replace or remove look-up table
The look-up table `NAT_REG_ID` defines geographical groups of countries (called "regions"). Furthermore, it contains a mapping from countries to impact function IDs which is only used together with the river flood module.

#### 2.1) Geographical grouping of countries
This particular grouping of countries is similar to the UN geoscheme (M49, https://en.wikipedia.org/wiki/List_of_countries_by_United_Nations_geoscheme), but not exactly the same. Furthermore, it is not in accordance with any other geographical grouping covered by the Python package `country-converter` (https://pypi.org/project/country-converter/).

The grouping seems to be of some interest for river flood simulation. I would suggest to make this clear in the naming of the variables and of the CSV-file. It should be clear that this data is somehow related to the river flood module.

#### 2.2) Impact function IDs
In `NAT_REG_ID`, there are 6 different impact function IDs according to the continental regions Africa, Asia, Europe, North America, Australia/Oceania, South America. We could easily list the countries for each ID directly in the Python code or use something like the `country-converter` package. The file `NAT_REG_ID` is not really the best place for that purpose - especially since the impact functions listed there are specific for the flood module.

**Comment by togeig** [comment]
Hi tovogt,
I am very much in favor of your efforts! The 150as and 360as files are a relict from the CLIMADA Matlab times and the first interaction with ISIMIP (not only floods, but also tropical cyclones). I think that they should remain available for the sake of reproducibility (e.g. the TCE-DAT archive) but they should not be the foundation for general and newer CLIMADA development and applications.

**Comment by tovogt** [comment]
As a first step, I renamed the `GLB_CENTROIDS_NC` constant to `ISIMIP_GPWV3_NATID_150AS` and added some documentation with references:
https://github.com/CLIMADA-project/climada_python/blob/3a3b16cde69ca651501b51e6726698e55d16b191/climada/util/constants.py#L56-L66
You are right, we should keep the data for future reference.

**Comment by tovogt** [comment]
As suggested by @togeig, all data will remain available for reproducibility. But docstrings have been added and more specific variable names (aliases) are now used in the code. In a last step, I renamed `NAT_REF_ID` to `RIVER_FLOOD_REGIONS_CSV`:
https://github.com/CLIMADA-project/climada_python/blob/09564712bd4721792e1743579b6ee11ab475d0ab/climada/util/constants.py#L89-L90
This solves the issue for me, because it turns out it was mostly about documentation and naming of variables/constants.
